src/algorithms/signal_source/gnuradio_blocks/fifo_reader.h

```diff
@@ -50,41 +49,16 @@ class FifoReader : virtual public gr::sync_block
     //! (gr handles this with public and private header pair)
     FifoReader(const std::string &file_name, const std::string &sample_type);
 
+    //! \brief Destructor
+    //! private destructor
+    ~FifoReader();
+
     size_t read_gr_complex(int noutput_items, gr_vector_void_star &output_items);
+
     //! function to read data out of FIFO which is stored as interleaved I/Q stream.
     //! template argument determines sample_type
     template <typename Type>
-    size_t read_interleaved(int noutput_items, gr_vector_void_star &output_items)
```

Comment:
Template definitions need to be visible at the point of instantiation. Most of the time, that means defining them in the header. It's possible that there is no harm, since this is a private method, but knowing that most developers aren't aware of these nuances, I prefer to comment at a minimum why it's safe to define this out of line.

An alternative to in-line definition is to move the definition after the class declaration in this file. That keeps the declaration easier to read, but satisfies the visibility requirements.

Reply:
Thanks. For observers, also see: https://stackoverflow.com/questions/495021/why-can-templates-only-be-implemented-in-the-header-file.